source/UF/SystemFNotionOfResizing.lagda

```diff
@@ -0,0 +1,348 @@
+--------------------------------------------------------------------------------
+title:        System F Resizing considered as an axiom
+authors:      ["Sam Speight", "Ayberk Tosun"]
+date-started: 2024-05-21
+--------------------------------------------------------------------------------
+
+This module contains some notes from various discussions with Sam Speight.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Subsingletons
+
+module UF.SystemFNotionOfResizing
+        (fe : Fun-Ext)
+        (pe : Prop-Ext)
+        (pt : propositional-truncations-exist)
+       where
+
+open import InjectiveTypes.Resizing
+open import MLTT.Spartan
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.Logic
+open import UF.NotNotStablePropositions
+open import UF.Retracts
+open import UF.Size
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+
+open AllCombinators pt fe
+
+\end{code}
+
+One can consider System F resizing in a universe polymorphic form, but we think
+this should be inconsistent due to some form of Girard’s Paradox. This is
+because it gives nested impredicative universes which is known to be
+inconsistent. However, there are lots of details to check here.
+
+\begin{code}
+
+Generalized-System-F-Resizing : (𝓤 𝓥 : Universe) → (𝓤 ⊔ 𝓥) ⁺  ̇
+Generalized-System-F-Resizing 𝓤 𝓥 =
+ (A : 𝓤 ⊔ 𝓥  ̇) → (B : A → 𝓤  ̇) → (Π x ꞉ A , B x) is 𝓤 small
+
+\end{code}
+
+TODO: prove that this generalized form is inconsistent.
+
+
+The special case of this notion of resizing where we pick `𝓤 := 𝓤₀` and
+`𝓥 := 𝓤₁` should be consistent.
+
+\begin{code}
+
+System-F-Resizing : 𝓤₂  ̇
```

Comment:
Should this really be called System F resizing? Is this the same form of impredicativity as in CoC?

Reply:
In any case, if we agree that this is the same as in System F, it probably should be called System F impredicativity, I think.